optuna/integration/mlflow.py

```diff
@@ -0,0 +1,134 @@
+import optuna
+from optuna._experimental import experimental
+from optuna import structs
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+
+try:
+    import mlflow
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    mlflow = object
+
+
+def _check_mlflow_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "MLflow is not available. Please install MLflow to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "mlflow`. For further information, please refer to the installation guide "
+            "of MLflow. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+@experimental("1.4.0")
+class MLflowCallback(object):
+    """Callback to track optuna trials with MLflow.
```

Comment:
Please be consistent about the use of upper/lower case of "Optuna," starting with the upper case.

Reply:
Fixed.